src/tool/xmeta/cpp/models/class_model.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <memory>
+#include <string_view>
+#include <vector>
+
+#include "base_model.h"
+#include "interface_model.h"
+#include "method_model.h"
+#include "property_model.h"
+#include "event_model.h"
+
+namespace xlang::xmeta
+{
+    // Note: All classes are implicitly public, and explicitly specifying this is not allowed.
+    enum class class_modifier_t
+    {
+        none,
+        sealed_class,
+        static_class
+    };
+
+    struct class_model : base_model
+    {
+        class_model(std::string_view const& id, size_t decl_line) : base_model{ id, decl_line } { }
+        class_model() = delete;
+
+        std::string_view class_base_id;
+        std::shared_ptr<class_model> class_base;
+        std::vector<std::string_view> interface_bases;
+        std::vector<std::string_view> type_parameters;
```

Comment:
It's not immediately obvious to me how type_parameters, formal_parameters, and generic_types interact.

Reply:
Ah, generic_types are the same as type_parameters and can be removed. formal_parameters are for methods and delegates, it shouldn't even be in this file. I'll remove the two (generic_types and formal_parameters).